### PR TITLE
[NMS] Event component improvements

### PR DESCRIPTION
**Pull request by karthiksubraveti**
## Summary
Made the following changes
- Improve event component to use multi stream event api
- Add histogram component to event component and enable search based on event types
- Made the event component pull remote data dynamically upon page change

## Test Plan
yarn test all succeeds.
Need to add a unit test to verify this component

event_component_gif

### CODE REVIEW

File: nms/app/fbcnms-packages/fbcnms-magma-api/bin/generateAPIFromSwagger.sh [hunk 1]
```diff
@@ -16,8 +16,12 @@ HEADER="/**
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
+ *
+ * @flow
+ * @generated
+ * This file is generated by generateAPIFromSwagger.sh .
```
Comment (andreilee) [inline nms/app/fbcnms-packages/fbcnms-magma-api/bin/generateAPIFromSwagger.sh:22]: Surprised these lines got added, implying that this file generated itself. What happened here?

File: nms/app/fbcnms-projects/magmalte/app/views/events/EventsTable.js [hunk 2]
```diff
@@ -78,12 +86,7 @@ function getEventDescription(event) {
 const streamNameMagmad = 'magmad';
 const streamNameSessiond = 'sessiond';
 
-export const magmaEventTypes = Object.freeze({
-  NETWORK: 1,
-  GATEWAY: 2,
-  SUBSCRIBER: 3,
-});
-
+export type magmaEventStream = 'NETWORK' | 'GATEWAY' | 'SUBSCRIBER';
```
Comment (andreilee) [inline nms/app/fbcnms-projects/magmalte/app/views/events/EventsTable.js:89]: 👍

File: nms/app/fbcnms-projects/magmalte/app/views/events/EventsTable.js [hunk 3]
```diff
@@ -137,85 +140,160 @@ function ExpandEvent(props: EventDescriptionProps) {
...
+      start: startDate,
+      end: endDate,
+      delta: delta,
+      unit: unit,
+      format: format,
     };
-  });
+  }, [startDate, endDate]);
+
+  function DateFilter() {
```
Comment (andreilee) [inline nms/app/fbcnms-projects/magmalte/app/views/events/EventsTable.js:235]: 👍

File: nms/app/fbcnms-projects/magmalte/app/views/events/EventsTable.js [hunk 4]
```diff
@@ -250,25 +338,55 @@ export default function EventsTable({
...
-                title: 'More Details',
-                field: 'eventDescription',
-                render: rowData => <ExpandEvent rowData={rowData} />,
-              },
-            ]}
-            options={{
-              actionsColumnIndex: -1,
-              pageSizeOptions: [10, 20],
-            }}
-          />
```
Comment (andreilee) [inline nms/app/fbcnms-projects/magmalte/app/views/events/EventsTable.js:271]: Why did you switch off the ActionTable?

File: nms/app/fbcnms-packages/fbcnms-magma-api/bin/generateAPIFromSwagger.sh (hunk 1 above)
Comment (karthiksubraveti) [reply nms/app/fbcnms-packages/fbcnms-magma-api/bin/generateAPIFromSwagger.sh:22]: No. I added these lines. Shruti had made few changes in headers and Magma Api bindings didn't generate properly. I added the flow header and added these lines.

File: nms/app/fbcnms-projects/magmalte/app/views/events/EventsTable.js (hunk 4 above)
Comment (karthiksubraveti) [reply nms/app/fbcnms-projects/magmalte/app/views/events/EventsTable.js:271]: I didn't turn it off, i moved it under the grid. The diff doesn't seem to show that properly.

**Review by xjtian — APPROVED** [review]

**Pull request merged**